### PR TITLE
rpm,deb: chown bluestore block* links to ceph.ceph on install/upgrade

### PR DESCRIPTION
**Pull request by liewegas**
This works around https://tracker.ceph.com/issues/24903

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
Is this the right way to fix the ownership on upgrade?

**Comment by smithfarm** [comment]
@liewegas (commenting only on RPM) The way you implemented this, the `chown` will be run unconditionally every time the `ceph-osd` package gets installed (i.e. both new install and on upgrades). From the looks of it, the number of `chown` calls will be equal to the number of OSDs times two, which should not add too much to the scriptlet runtime.

The `ceph` user and group are guaranteed to exist, because they are created by the `%pre` scriptlet of `ceph-common` - i.e. before this scriptlet gets run.

If there was a concern that executing the `chown` on new install is really bad and undesirable, you could put a conditional block around it:

```
if [ $1 -eq 2 ] ; then
  chown ceph.ceph /var/lib/ceph/osd/*/block{.db,.wal} 2>& /dev/null || :
fi
```

but I don't think this is necessary. On new install the `chown` will just fail because no OSDs will have been deployed at that point, and IMO it's better to run it anyway, just to be sure.

@tserong Do you agree?

File: debian/ceph-osd.postinst [hunk 1]
```diff
@@ -25,6 +25,8 @@ case "$1" in
     configure)
 	[ -x /etc/init.d/procps ] && invoke-rc.d procps restart || :
 	[ -x /sbin/start ] && start ceph-osd-all || :
+	# work around https://tracker.ceph.com/issues/24903
+	chown ceph.ceph /var/lib/ceph/osd/*/block{.db,.wal} 2>& /dev/null || :
```
Comment (tchaikov) [inline debian/ceph-osd.postinst]: should be `ceph:ceph` ?

Comment (tchaikov) [reply debian/ceph-osd.postinst]: also `{.db,.wal}` is bashism, which does not work with dash.

**Comment by tserong** [comment]
> @tserong Do you agree?

Yup, IMO your reasoning is sound, and this change LGTM.

**Comment by b-ranto** [comment]
I am just wondering if this have enough impact to have this ~forever in the packaging. Especially since a simple reboot should fix this automatically. How many actual releases did this affect?

**Comment by liewegas** [comment]
Only the broken/unannounced 12.2.6 release.  We could leave it in luminous and remove it in later releases.

**Comment by b-ranto** [comment]
+1 for luminous only.

**Comment by smithfarm** [comment]
Instead of merging it to master, backporting to luminous, and reverting in master, would it be simpler to retarget this PR to luminous?

**Comment by jdurgin** [comment]
already merged to luminous in https://github.com/ceph/ceph/pull/23039

**Pull request closed without merging**